recipes/default.rb

```diff
@@ -61,8 +61,11 @@
     action :append
   end
 
+  ipaddress = node['hostname_cookbook']['hostsfile_ip']
+  ipaddress = node['ipaddress'] if ipaddress == 'ipaddress'
+
   hostsfile_entry 'set hostname' do
-    ip_address node['hostname_cookbook']['hostsfile_ip']
+    ip_address ipaddress
```

Comment:
If the node variable name wasn't so long, we could make this a clean one line conditional statement. But as-is, I'd say we shouldn't reassign `ipaddress` as a variable unnecessarily. 

```ruby
ip_address if node['hostname_cookbook']['hostsfile_ip'] == 'ipaddress'
             node['ipaddress']
           else
             node['hostname_cookbook']['hostsfile_ip']
           end
```